src/dynamics/periodic_pts.lean

```diff
@@ -283,6 +283,13 @@ lemma is_periodic_pt_iff_minimal_period_dvd :
 
 open nat
 
+lemma minimal_period_eq_minimal_period_iff {g : β → β} {y : β} :
+  minimal_period f x = minimal_period g y ↔ ∀ n, is_periodic_pt f n x ↔ is_periodic_pt g n y :=
+begin
+  simp_rw is_periodic_pt_iff_minimal_period_dvd,
+  exact ⟨λ h n, by rw h, λ h, nat.dvd_antisymm ((h _).mpr (dvd_refl _)) ((h _).mp (dvd_refl _))⟩,
+end
```

Comment:
This lemma should probably be about natural numbers being equal iff they divide the same numbers.

If Yoneda's lemma is in the category theory part of the library, we might prove this using the partial order under divisibility as a category on nat. (-;

Reply:
I agree. Although I'm pretty sure some people won't appreciate if we import category theory into data.nat.basic :-)

For completeness, we should also include the dual version.

Reply:
Ok, I will extract the `nat` lemma: we will leave the category theory taking over every statement in mathlib for another day!